jupyterlab/commands.py

```diff
@@ -1662,6 +1686,35 @@ def _node_check(logger):
         msg = 'Please install nodejs %s before continuing. nodejs may be installed using conda or directly from the nodejs website.' % ver
         raise ValueError(msg)
 
+def _yarn_config(logger):
+    """Get the yarn configuration.
+    
+    Returns
+    -------
+    {"yarn config": dict, "npm config": dict} if unsuccessfull the subdictionary are empty
+    """
+    node = which('node')
+    try:
+        output_binary = subprocess.check_output([node, YARN_PATH, 'config', 'list', '--json'], cwd=HERE)
+        output = output_binary.decode('utf-8')
+        configuration = {}
+        lines = iter(output.splitlines())
+        try:
+            for line in lines:
+                info = json.loads(line)
+                if info["type"] == "info":
+                    key = info["data"]
+                    inspect = json.loads(next(lines))
+                    if inspect["type"] == "inspect":
+                        configuration[key] = inspect["data"]
+        except StopIteration:
+            pass
+        logger.debug("Yarn configuration loaded.")
+        return configuration
+    except Exception as e:
+        logger.error("Fail to get yarn configuration. {!s}".format(e))
```

Comment:
If the exception is a `CalledProcessError` (from the subprocess call) we should ideally also log the captured output (`error.output`).

Reply:
Thanks for the tip.